### DISCUSSION TITLE
Flexible/Customizable keystrokes

### DISCUSSION THREAD

**Discussion by Cookiesmuch** [ppy/osu#20423] [Ideas]
Since the first few fully-functioning releases of Osu!Lazer, the keystrokes in the bottom-right hand corner of the screen, has been very customizable and unflexible. For starters, even is !Stable the keystrokes has been changeable with skins etc., considering you can change most of the UI elements in !Lazer are editable and are flexible in their appearance, this is one of the biggest outliers.

The bigger thing I have thought is more of a support for a niche play style. 
In !Stable, you have four keystroke (button/indicator?)s, what if those weren't limited to just Key1 Key2 Mb1 and Mb2? 
Other rulesets (Mania and custom rulesets) already implement more keystrokes. 
We have your standard single tapping, and your double tapping, but I've been slowly transitioning to what you could simply call triple tapping. This is the (to state the obvious) act of using 3 fingers to tap, currently this playstyle in my hands are erratic but I would imagine if someone got used to it, it would be an elegant zxc-zxc-zxc or a xc-xz-xc-xz-xc.
Coming off of that small sidetrack, what if players could customize the amount (and assigned key) of keystrokes? What if we're not just limited to assigning Key1 and Key2, then having 2 keystrokes?

**Comment by Cookiesmuch** [comment]
Edit: I know the 7K mod idea _might_ fix this, assuming it shows all 7 keys, however making the keystrokes more flexible would also aid in making more features like the 7K mod easier, no? As well as a small flexibility upgrade to key bind settings. Right now I just bind two keys to key bind 1 and the last key and Mb1 to Key2.

**Reply by ghost** [reply]
I think you're misunderstanding how the keys and mouse work on stable. If you're holding K1, any clicks to M1 won't be registered (i.e. you can not press K1 and M1 at the same time). 

I googled "multiple keybind osu" and found some pretty interesting discussions. Highly recommend you read some of them

**Reply by ghost** [reply]
also the 7k mod is for mania, and there are already 7k mania beatmaps available for play.

**Reply by Cookiesmuch** [reply]
I was talking about the 7K mod discussion for normal Osu (the idea was to allow 7 keys used instead of 2 (4 if you include the mouse slots in lazer) I know that using more than one key in normal osu is well established, but having keystrokes for this would be great :D. Thank you for the comment, I hope that adding more keys to normal osu would be made possible, and the in-game keystrokes could then properly support my triple-tapping!
